jdaviz/configs/cubeviz/plugins/spectral_extraction/tests/test_spectral_extraction.py

```diff
@@ -520,9 +520,18 @@ def test_spectral_extraction_with_correct_sum_units(cubeviz_helper,
     cubeviz_helper.load_data(spectrum1d_cube_fluxunit_jy_per_steradian)
     spec_extr_plugin = cubeviz_helper.plugins['Spectral Extraction']._obj
     collapsed = spec_extr_plugin.extract()
+
+    assert '_pixel_scale_factor' in collapsed.meta
+
+    # Original units in Jy / sr
+    expected_flux_values = [190., 590., 990., 1390., 1790., 2190., 2590., 2990., 3390., 3790.]
+    # After collapsing, sr is removed via the scale factor and the extracted spectrum is in Jy
+    expected_flux_values = [flux * collapsed.meta.get('_pixel_scale_factor')
+                            for flux in expected_flux_values]
```

Comment:
can this be done with array math instead of a list comprehension?

Reply:
I have now made an update to make the `expect_flux_values` is an array instead of a list, this will remove the list comprehension all together